src/Instrumentation/ReactHTTP/src/ReactHttpInstrumentation.php

```diff
@@ -0,0 +1,161 @@
+                    ->setAttribute(TraceAttributes::SERVER_ADDRESS, $request->getUri()->getHost())
+                    ->setAttribute(TraceAttributes::SERVER_PORT, $request->getUri()->getPort())
+                    ->setAttribute(TraceAttributes::URL_FULL, (string) $request->getUri())
+                    ->setAttribute(TraceAttributes::NETWORK_PROTOCOL_NAME, 'http')
+                    ->setAttribute(TraceAttributes::NETWORK_PROTOCOL_VERSION, $request->getProtocolVersion())
+                    ->setAttribute(TraceAttributes::HTTP_REQUEST_BODY_SIZE, $request->getHeaderLine('Content-Length'))
+                    ->setAttribute(TraceAttributes::URL_SCHEME, $request->getUri()->getScheme())
+                    ->setAttribute(TraceAttributes::USER_AGENT_ORIGINAL, $request->getHeaderLine('User-Agent'))
+                    // https://opentelemetry.io/docs/specs/semconv/code/
+                    ->setAttribute(TraceAttributes::CODE_FUNCTION_NAME, sprintf('%s::%s', $class, $function))
+                    ->setAttribute(TraceAttributes::CODE_FILE_PATH, $filename)
+                    ->setAttribute(TraceAttributes::CODE_LINE_NUMBER, $lineno);
+
+                /**
+                 * This cannot be tested with PHPUnit without modifying php.ini with:
+                 * otel.instrumentation.http.request_headers[]="Accept"
+                 */
+                // @codeCoverageIgnoreStart
+                foreach ((array) (get_cfg_var('otel.instrumentation.http.request_headers') ?: []) as $header) {
+                    if ($request->hasHeader($header)) {
+                        $spanBuilder->setAttribute(
+                            sprintf('%s.%s', TraceAttributes::HTTP_REQUEST_HEADER, strtolower($header)),
+                            $request->getHeaderLine($header)
+                        );
+                    }
+                }
+                // @codeCoverageIgnoreEnd
+
+                $span = $spanBuilder->startSpan();
+                $context = $span->storeInContext($parentContext);
+                $propagator->inject($request, HeadersPropagator::instance(), $context);
+
+                Context::storage()->attach($context);
+
+                return [$request];
+            },
+            post: static function (Transaction $transaction, array $params, PromiseInterface $promise, ?Throwable $exception): PromiseInterface {
+                $scope = Context::storage()->scope();
+                $scope?->detach();
+
+                if (!$scope || $scope->context() === Context::getCurrent()) {
+                    return $promise;
+                }
+
+                $span = Span::fromContext($scope->context());
+
+                return $promise->then(
+                    onFulfilled: function (ResponseInterface $response) use ($span) {
+                        $span
+                            ->setAttribute(TraceAttributes::HTTP_RESPONSE_STATUS_CODE, $response->getStatusCode())
```

Comment:
Should set span status to error if status codes `>= 400`.

Reply:
I tested this and there's no way to receive a status code >=400 without ReactPHP calling `onRejected` instead of `onFulfilled`, with a `ResponseException` wrapping the response.

Reply:
Should be possible with `Browser::withRejectErrorResponse(false)`, see https://github.com/reactphp/http/blob/fc5f0a6739ba4a10de3847bab4688c602e5fe948/src/Browser.php#L525-L538
> You can pass in a bool `false` to disable rejecting incoming responses that use a 4xx or 5xx response status code.

Reply:
Gah, good catch. Handling and testing in the next commit.